src/libs/ReportUtils.ts

```diff
@@ -5890,6 +5914,84 @@ function buildOptimisticChangePolicyReportAction(fromPolicyID: string | undefine
     };
 }
 
+/**
+ * Builds an optimistic MOVED_TRANSACTION report action with a randomly generated reportActionID.
+ * This action is used when we change the workspace of a report.
+ */
+function buildOptimisticMovedTransactionAction(oldReportID: string | undefined, toReportID: string, transactionID: string): ReportAction {
+    const originalMessage = {
+        oldReportID,
+        toReportID,
+        transactionID,
+    };
+
+    const changePolicyReportActionMessage = [
+        {
+            type: CONST.REPORT.MESSAGE.TYPE.TEXT,
+            text: `moved this expense to [Expense Report ${toReportID}]`,
+        },
+    ];
+
+    return {
+        actionName: CONST.REPORT.ACTIONS.TYPE.MOVED_TRANSACTION,
+        reportID: oldReportID,
+        actorAccountID: currentUserAccountID,
+        avatar: getCurrentUserAvatar(),
+        created: DateUtils.getDBTime(),
+        originalMessage,
+        message: changePolicyReportActionMessage,
+        person: [
+            {
+                style: 'strong',
+                text: getCurrentUserDisplayNameOrEmail(),
+                type: 'TEXT',
+            },
+        ],
+        reportActionID: rand64(),
```

Comment:
We'll need to add this ID to the `reportActionIDToThreadReportIDMap` param we send to the API
```suggestion
        reportActionID: rand64(),
```